File: src/test/java/se/johannesdahlgren/adventofcode2021/Day7Test.java
package se.johannesdahlgren.adventofcode2021;


import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.Test;

class Day7Test {

  @Test
  void part1Example() {
    Day7 day7 = new Day7("day7example.txt");
    long fuelRequired = day7.cheapestAlign();
    assertThat(fuelRequired).isEqualTo(37L);
  }

  @Test
  void part1() {
    Day7 day7 = new Day7("day7.txt");
    long fuelRequired = day7.cheapestAlign();
    assertThat(fuelRequired).isEqualTo(344138L);
  }

  @Test
  void part2Example() {
    Day7 day7 = new Day7("day7example.txt");
    long fuelRequired = day7.cheapestAlignIncreasingBurnRate();
    assertThat(fuelRequired).isEqualTo(168L);
  }

  @Test
  void part2() {
    Day7 day7 = new Day7("day7.txt");
    long fuelRequired = day7.cheapestAlignIncreasingBurnRate();
    assertThat(fuelRequired).isEqualTo(94862124L);
  }

}

File: src/test/kotlin/se/johannesdahlgren/aoc22/Day7Test.kt
package se.johannesdahlgren.aoc22

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class Day7Test {
    @Test
    fun part1Example() {
        val day7 = Day7("day7Example")
        assertThat(day7.calcTotalSize(100000)).isEqualTo(95437)
    }

    @Test
    fun part1() {
        val day7 = Day7("day7")
        assertThat(day7.calcTotalSize(100000)).isEqualTo(1206825)
    }

    @Test
    fun part2Example() {
        val day7 = Day7("day7Example")
        assertThat(day7.findSmallestFolder(30000000)).isEqualTo(24933642)
    }

    @Test
    fun part2() {
        val day7 = Day7("day7")
        assertThat(day7.findSmallestFolder(30000000)).isEqualTo(9608311)
    }
}